x-pack/plugin/vector-tile/src/main/java/org/elasticsearch/xpack/vectortile/rest/RestVectorTileAction.java

```diff
@@ -265,12 +266,10 @@ private static SearchRequestBuilder searchRequestBuilder(RestCancellableNodeClie
                 featureBuilder.mergeFrom((byte[]) feature);
                 VectorTileUtils.addPropertyToFeature(featureBuilder, layerProps, ID_TAG, searchHit.getId());
                 VectorTileUtils.addPropertyToFeature(featureBuilder, layerProps, INDEX_TAG, searchHit.getIndex());
-                if (fields != null) {
-                    for (FieldAndFormat field : fields) {
-                        final DocumentField documentField = searchHit.field(field.field);
-                        if (documentField != null) {
-                            VectorTileUtils.addPropertyToFeature(featureBuilder, layerProps, field.field, documentField.getValue());
-                        }
+                final Map<String, DocumentField> fields = searchHit.getFields();
```

Comment:
I presume `searchHits.getFields()` contains the subset of `request.getFieldAndFormats()` that was actually found? So this way we skip fields not found?

Reply:
I added a test for a non-existing field and for a field with null value. Everything seems to behave as expected